lib/errorResponseInterceptor.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * © Copyright IBM Corporation 2024. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function errorResponseInterceptor(axiosError) {
+  if (
```

Comment:
I am confused on inconsistent use of optional chain on `axiosError` in this `if`. Sometimes it used, sometime it is missing and anyway, if we expect `axiosError` to be able to be `nil`, it'd be cleaner to check and return promise reject as a first check and then change condition to use it with confidence that it is present.

Reply:
I left optionals by a coincidence. Personally, I do not like using them, those were used in the proposal, so I think we should ask @ricellis' opinion about this. I do not think an `axiosError` would be `nil`, so I would not cover that, but LMK if I miss an edge case here.

Anyways, I got rid off the optionals in 3453e093.

Reply:
Yeah it's fine, the prototype was a quick experiment.